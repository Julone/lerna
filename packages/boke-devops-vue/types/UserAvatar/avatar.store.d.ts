export declare const useCustomProps: () => {
    NAME_KEY: import("vue").Ref<any>;
    AVATAR_KEY: import("vue").Ref<any>;
    DEPT_KEY: import("vue").Ref<any>;
    USERID_KEY: import("vue").Ref<any>;
    global_always_avatar_is_hihglight: any;
};
