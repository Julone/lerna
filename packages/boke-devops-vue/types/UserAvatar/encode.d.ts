export declare function hex_sha1(s: any): string;
