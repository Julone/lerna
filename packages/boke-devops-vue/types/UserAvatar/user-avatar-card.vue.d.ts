declare const _default: import("vue").DefineComponent<Readonly<{
    userinfo?: any;
}>, any, {
    showViewer: boolean;
    spacer: import("vue").VNode<import("vue").RendererNode, import("vue").RendererElement, {
        [key: string]: any;
    }>;
    isDark: boolean;
    extraInfo: null;
    seeAllProjects: boolean;
    seeAllRoles: boolean;
}, {
    projectslist(): any;
    projectslistCount(): any;
    extraInfogroups(): any;
    extraInfogroupsCount(): any;
}, {
    onChat(account: any): void;
}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {}, string, import("vue").PublicProps, Readonly<import("vue").ExtractPropTypes<Readonly<{
    userinfo?: any;
}>>>, {
    readonly userinfo?: any;
}, {}>;
export default _default;
