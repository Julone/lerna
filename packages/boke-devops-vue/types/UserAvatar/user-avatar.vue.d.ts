declare const _default: import("vue").DefineComponent<{
    size: {
        type: (StringConstructor | NumberConstructor)[];
        required: false;
        default: number;
    };
    hidePopover: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    onlyIcon: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    onlyName: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    noID: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    linkType: {
        type: null;
        required: false;
        default: string;
    };
    data: {
        type: null;
        required: false;
        default: null;
    };
    shape: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    disabledPopover: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    tag: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    closable: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    color: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    disabled: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    suffix: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    is_highlight: {
        type: FunctionConstructor;
        required: false;
        default: () => boolean;
    };
}, {
    user_name: import("vue").ComputedRef<any>;
    dept_name: import("vue").ComputedRef<any>;
    user_id: import("vue").ComputedRef<any>;
    data: import("vue").Ref<any>;
    user_icon: import("vue").Ref<any>;
    isLoading: import("vue").Ref<boolean>;
    isHigh: () => any;
    props: import(".pnpm/@vue+shared@3.4.31/node_modules/@vue/shared").LooseRequired<{
        readonly size: string | number;
        readonly data: any;
        readonly color: string;
        readonly closable: boolean;
        readonly shape: string;
        readonly disabled: boolean;
        readonly hidePopover: boolean;
        readonly onlyIcon: boolean;
        readonly onlyName: boolean;
        readonly noID: boolean;
        readonly linkType: any;
        readonly disabledPopover: boolean;
        readonly tag: boolean;
        readonly suffix: string;
        readonly is_highlight: Function;
        readonly onClose?: ((...args: any[]) => any) | undefined;
    } & {}>;
    attrs: {
        [x: string]: unknown;
    };
    emit: (event: "close", ...args: any[]) => void;
    onClose: () => void;
    global_disabled_avatar_card: boolean;
    targetRef: import("vue").Ref<any>;
    avatarRef: import("vue").Ref<any>;
    popoverVisible: import("vue").Ref<boolean>;
}, any, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, "close"[], "close", import("vue").PublicProps, Readonly<import("vue").ExtractPropTypes<{
    size: {
        type: (StringConstructor | NumberConstructor)[];
        required: false;
        default: number;
    };
    hidePopover: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    onlyIcon: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    onlyName: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    noID: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    linkType: {
        type: null;
        required: false;
        default: string;
    };
    data: {
        type: null;
        required: false;
        default: null;
    };
    shape: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    disabledPopover: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    tag: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    closable: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    color: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    disabled: {
        type: BooleanConstructor;
        required: false;
        default: boolean;
    };
    suffix: {
        type: StringConstructor;
        required: false;
        default: string;
    };
    is_highlight: {
        type: FunctionConstructor;
        required: false;
        default: () => boolean;
    };
}>> & {
    onClose?: ((...args: any[]) => any) | undefined;
}, {
    size: string | number;
    data: any;
    color: string;
    closable: boolean;
    shape: string;
    disabled: boolean;
    hidePopover: boolean;
    onlyIcon: boolean;
    onlyName: boolean;
    noID: boolean;
    linkType: any;
    disabledPopover: boolean;
    tag: boolean;
    suffix: string;
    is_highlight: Function;
}, {}>;
export default _default;
