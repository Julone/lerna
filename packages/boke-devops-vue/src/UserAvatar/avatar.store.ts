import { computed, toRef,  getCurrentInstance, inject } from "vue"
import { globalConfig, globalState } from "../Provider/store"
import { mergeWith } from "lodash-es"

export const useCustomProps = () => {
    const props = getCurrentInstance()?.attrs as any;
    const injectGlobalConfig = inject("globalConfig") as any
    const injectGlobalState = inject("globalState") as any
    const mergedProps = mergeWith({}, globalConfig.avatar, injectGlobalConfig?.avatar, props?.customProps || {});
    const mergedAvatarCardProps = mergeWith({}, globalConfig.avatarCard, injectGlobalConfig?.avatarCard);


    let NAME_KEY = toRef(mergedProps, 'user_name')
    let AVATAR_KEY = toRef(mergedProps, 'user_avatar')
    let DEPT_KEY = toRef(mergedProps, 'dept_name')
    let USERID_KEY = toRef(mergedProps, 'user_id')


    const global_always_avatar_is_hihglight = injectGlobalState?.global_always_avatar_is_hihglight || globalState?.global_always_avatar_is_hihglight
    const global_disabled_avatar_card = injectGlobalConfig?.avatarCard?.enabled == false
    const AVATAR_CARD_URL = mergedAvatarCardProps.api_url

    return {
        NAME_KEY,
        AVATAR_KEY,
        DEPT_KEY,
        USERID_KEY,
        global_always_avatar_is_hihglight,
        global_disabled_avatar_card,
        AVATAR_CARD_URL
    }
}