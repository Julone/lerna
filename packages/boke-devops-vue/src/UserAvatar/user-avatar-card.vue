<template>
    <div
      class="mps-user-avator-rich-conent"
      style="display: flex; gap: 16px; flex-direction: row;     align-items: flex-start;"
    >
      <div style="z-index: 2">
        <img :src="userinfo.user_icon" style="width: 70px; height: 70px;border-radius: 500px;margin-top: 10px;" />
      </div>
      <div style="z-index: 2">
        <p
          class="demo-rich-content__name"
          style="margin: 0 0 6px 0; font-weight: 500"
        >
          <a-space :size="12" :spacer="spacer" :wrap="false">
            <span style="font-weight:bold">{{ userinfo.user_name }}</span>
            <span class="dept_class">{{ userinfo.dept_name }}</span>
            <span class="dept_class" v-if="userinfo.dept_name != extraInfo?.min_fs_department_name && extraInfo?.min_fs_department_name">{{ extraInfo?.min_fs_department_name }}</span>
          </a-space>
        </p>
        <p
            v-if="extraInfo"
          class="demo-rich-content__mention"
          style="
            margin: 0 0 8px 0;
            font-size: 14px;
            color: var(--el-color-info);
          "
        >
          <a-space :wrap="true" :size="2">
            <ATag :bordered="false" color="orange" v-for="el in extraInfogroups" :key="el.name" size="small">
              {{el.name}}
            </ATag>
            <a-button  type="link" size="small"  @click="seeAllRoles = !seeAllRoles, $emit('pinCard')" style="zoom: 0.8">关联角色({{extraInfogroupsCount}})<CaretDownOutlined :rotate="seeAllRoles?180: 0" /></a-button>
          </a-space>
        </p>
        <p
            v-if="extraInfo"
            class="demo-rich-content__mention"
            style="
            margin: 0 0 6px 0;
            font-size: 14px;
            color: var(--el-color-info);
          "
        >
          <a-space :wrap="true" :size="2">
            <MpsRoleTag v-for="(el,index) in projectslist"  :key="el" size="small">
              {{el}}
            </MpsRoleTag>
            <a-button  type="link" size="small"  @click="seeAllProjects = !seeAllProjects, $emit('pinCard')" style="zoom: 0.8">关联项目({{projectslistCount}})<CaretDownOutlined :rotate="seeAllProjects?180: 0" /></a-button>
          </a-space>
        </p>
        <p  class="demo-rich-content__desc" style="margin:  8px 0 0">
          <a-space :size="2" :spacer="spacer">
            <a-button
                type="primary"
              size="small"
                :disabled="!extraInfo"
              @click="onChat(extraInfo?.fs_open_id)"
            >
              飞书联系
            </a-button>
          </a-space>
        </p>
        <a style="position:absolute; right: 8px; bottom: 8px;float:right; font-size: 12px; opacity: 0.8; color: #ccc; transform: translateY(0px)" href="https://devops.pocketcity.com/">数据来源: Aegis</a>

      </div>
      <div
        class="mps-user-avator-rich-conent_after"
        :style="{
          'background-image': `url('https://sf3-scmcdn-cn.feishucdn.com/obj/feishu-static/lark/open/app_store/images/banner-2c2539b4.png')`,
          opacity: 0.4,
          'background-size': 'cover',
          'background-position-y': '-30px',
        }"
      ></div>
    </div>
</template>
<style lang="scss" scoped>
  .mps-user-avator-rich-conent {
    min-width: 270px;
    max-width: 350px;
    :deep(.el-avatar) {
      img {
        width: 100%;
        cursor: zoom-in;
      }
    }
    :deep(.dept_class){
      color: #333;
      font-size: 12px;
    }

    .mps-user-avator-rich-conent_after {
      content: '';
      position: absolute;
      right: 0;
      bottom: 0;
      left: 0;
      top: 0px;
      z-index: 1;
      border-radius: 8px;

      background-repeat: no-repeat;
      background-size: contain;
      background-position: left top;
    }
  }
</style>
<script>
  import { defineComponent, createVNode } from 'vue'
  import {Tag as ATag ,Divider as ADivider, Space as ASpace, Button as AButton} from "ant-design-vue"
  import {hex_sha1} from "./encode";
  import {globalConfig}  from "./../Provider/store"
  import {CaretDownOutlined} from "@ant-design/icons-vue"
  export default defineComponent({
    props: ['userinfo'],
    components: { MpsRoleTag: ATag,  MpsGroupTag: ATag, MpsDictTag: ATag ,ASpace,AButton,ATag,CaretDownOutlined},
    emit: ['pinCard'],
    data() {
      return {
        showViewer: false,
        spacer: createVNode(ADivider, {
          direction: 'vertical',
          style: { margin: '0 2px' },
        }),
        isDark: false,
        extraInfo: null,
        seeAllProjects:  false,
        seeAllRoles: false
      }
    },
    computed: {
      projectslist(){
        return (this.extraInfo?.projects_list || []).filter((item, index) => {
          if(this.seeAllProjects) {
            return true
          }
          return index < 4
        })
      },
      projectslistCount(){
        return (this.extraInfo?.projects_list || [])?.length
      },
      extraInfogroups(){
        return (this.extraInfo?.groups || []).filter((item, index) => {
          if(this.seeAllRoles) {
            return true
          }
          return index < 4
        })
      },
      extraInfogroupsCount(){
        return (this.extraInfo?.groups || []).length
      }

    },
    methods: {
      onChat(account) {
        window.open(`feishu://applink.feishu.cn/client/chat/open?openId=${account}`)
      },
    },
    mounted() {
      // 预备参数
      const jsapi_ticket = Math.random().toString(36).substr(2, 18)
      const noncestr = Math.random().toString(36).substr(2, 7)
      const user_id =  this.userinfo?.user_id
      const timestamp = Date.now()
      // 加密参数生成签名
      const encode = `jsapi_ticket=${jsapi_ticket}&noncestr=${noncestr}&user_id=${user_id}&timestamp=${timestamp}`
      const signature = hex_sha1(encode)
      // 查询
      const api = `/users/getUserInfo/?${encode}&signature=${signature}`
      fetch(globalConfig.avatarCard.api_url + api,).then(res=> res.json()).then(res=> {
        if (res.code == 0) {
          this.extraInfo = res.data
        }
      })
    },
  })
</script>

