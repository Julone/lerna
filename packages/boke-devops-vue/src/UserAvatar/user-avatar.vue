<template>
  <a-popover placement="bottomLeft" v-model:open="popoverVisible" :arrow="false" :mouseEnterDelay="0.1"
    :mouseLeaveDelay="0.3" transitionName="ant-zoom-big"
    :trigger="props.disabledPopover || props.disabled || global_disabled_avatar_card ? 'contextmenu' : 'hover'"
    :overlayStyle="{ zIndex: '9999', }">
    <template #content>
      <userAvatarCard :userinfo="{ user_name, dept_name, user_id, user_icon }" @pinCard="popoverVisible = true">
      </userAvatarCard>
    </template>
    <div style="display:inline-flex" ref="avatarRef">
      <ATag class="boke-useravator-wrapper" :class="{
        'wrapper-only-name': props.onlyName,
        is_tag: props.tag && !props.color,
        is_highlight: isHigh(),
        disabled: props.disabled,
        'only-icon-style': props.onlyIcon
      }" v-bind="attrs" :style="{ padding: '2px', paddingRight: '12px', borderRadius: '50px' }" v-if="data"
        :color="props.color || 'default'" :title="data?.[DEPT_KEY]">
        <div class="left-icon" ref="targetRef">
          <AAvatar v-if="!props.onlyName" :src="user_icon" :class="'avator-icon'" :size="props.size"></AAvatar>
          <span v-else>
            <span v-if="props.linkType == 'text'">
              {{ user_name }}
            </span>
            <el-link v-else :type="props.linkType" :underline="false">
              {{ user_name }}
            </el-link>
          </span>
        </div>
        <slot name="prefix"></slot>
        <div class="label" v-if="!props.onlyIcon && !props.onlyName">
          <span class="realname">
            {{
              isLoading
                ? "加载中..."
                : user_name
            }}
          </span>
          <span class="id" v-if="!props.noID">
            <span>{{ dept_name }}</span>
          </span>
        </div>
        <div v-if="props.suffix">{{ props.suffix }}</div>
        <AButton size="small" v-if="props.closable" @click.stop.prevent="onClose" :color="props.color" type="text"
          shape="circle" style="margin-left: 3px; transform: translateY(0) scale(0.6); opacity: 0.8">
          <template #icon>
            <CloseOutlined />
          </template>
        </AButton>
        <slot name="default" :scoped="data"></slot>
      </ATag>
    </div>
  </a-popover>
</template>

<script lang="jsx">
import { defineComponent } from "vue"
import {
  ref,
  defineEmits,
  computed,
} from "vue";
import { Avatar as AAvatar, Button as AButton, Tag as ATag, Popover as APopover, BadgeRibbon as ABadgeRibbon } from "ant-design-vue/es"
import { useCustomProps } from "./avatar.store"
import { isFunction } from "lodash-es"
import { CloseOutlined } from "@ant-design/icons-vue"
import userAvatarCard from "./user-avatar-card.vue";

export default defineComponent({
  inheritAttrs: false,
  name: "BokeUserAvator", // 切记：defineOptions() vue3.3才有，请使用defineComponent
  components: {
    'AAvatar': AAvatar,
    "AButton": AButton,
    "ATag": ATag,
    CloseOutlined,
    APopover,
    userAvatarCard,
    'ABadgeRibbon': ABadgeRibbon
  },
  props: {
    size: {
      type: [Number, String],
      required: false,
      default: 28
    },
    hidePopover: {
      type: Boolean,
      required: false,
      default: false
    },
    onlyIcon: {
      type: Boolean,
      required: false,
      default: false
    },
    onlyName: {
      type: Boolean,
      required: false,
      default: false
    },
    noID: {
      type: Boolean,
      required: false,
      default: false
    },
    linkType: {
      type: null,
      required: false,
      default: "primary"
    },
    data: {
      type: null,
      required: false,
      default: null
    },
    shape: {
      type: String,
      required: false,
      default: "circle"
    },
    disabledPopover: {
      type: Boolean,
      required: false,
      default: false
    },
    tag: {
      type: Boolean,
      required: false,
      default: false
    },
    closable: {
      type: Boolean,
      required: false,
      default: false
    },
    color: {
      type: String,
      required: false,
      default: ""
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false
    },
    suffix: {
      type: String,
      required: false,
      default: ""
    },
    is_highlight: {
      type: Function,
      required: false,
      default: () => false
    }
  },
  emits: ['close'],
  setup(props, { attrs, emit }) {
    const { NAME_KEY, AVATAR_KEY, DEPT_KEY, USERID_KEY, global_always_avatar_is_hihglight, global_disabled_avatar_card } = useCustomProps()
    const appVersion = "1.0.0";
    const targetRef = ref();
    const showViewer = ref(false);

    const isMyself = ref(false);

    const user_name = computed(() => {
      return props.data[NAME_KEY.value] || "未知用户";
    });

    const dept_name = computed(() => {
      return props.data[DEPT_KEY.value] || "未知部门";
    });

    const user_id = computed(() => {
      return props.data[USERID_KEY.value] || "未知用户";
    });
    const data = ref(props.data);
    const user_icon = ref(props.data[AVATAR_KEY.value]);
    const isLoading = ref(false);

    const isHigh = () => {
      if (global_always_avatar_is_hihglight) {
        return isFunction(global_always_avatar_is_hihglight) ? global_always_avatar_is_hihglight(data.value) : global_always_avatar_is_hihglight
      }
      return isFunction(props.is_highlight) ? props?.is_highlight(data.value) : props.is_highlight
    }

    const onClose = () => {
      emit("close")
    }
    const avatarRef = ref()
    const getPopupContainer = () => {
      console.log(avatarRef.value, "targetRef.value")
      return avatarRef.value
    }
    const popoverVisible = ref(false)
    return {
      user_name,
      dept_name,
      user_id,
      data,
      user_icon,
      isLoading,
      isHigh,
      props,
      attrs,
      emit,
      onClose,
      global_disabled_avatar_card,
      targetRef,
      avatarRef,
      popoverVisible
    }
  }
})
</script>
<style lang="less" scoped>
.boke-useravator-wrapper {
  display: inline-flex;
  position: relative;
  align-items: center;
  overflow: hidden;
  margin-left: 4px;
  border: none;

  &.ant-tag-default {
    cursor: default;
    background-color: transparent;

    &:hover {
      filter: brightness(0.8)
    }
  }

  &.is_tag {
    background-color: var(--boke-fill-color-light);
    border-radius: 118px;
  }

  &.is_highlight {
    border-radius: 118px;
    background-color: var(--boke-color-primary);
    color: white;
  }

  &.disabled {
    pointer-events: none;
  }

  .left-icon {
    display: inline-flex;
    padding: 0 4px 0 2px;
    position: relative;
  }

  &.wrapper-only-name {
    margin-left: 0px;

    .left-icon {
      padding: 0 0px;
    }
  }

  .label {
    font-size: 12px;
    white-space: nowrap;
    display: flex;
    flex-direction: column;
    max-width: 80px;
    margin-left: 4px;

    span {
      text-overflow: ellipsis;
      overflow: hidden;
      line-height: 14px;

      &.realname {
        margin-bottom: 0;

        &.lineThrough {
          // text-decoration: line-through;
          color: #75757599;
        }
      }

      &.id {
        opacity: 0.7;
        zoom: 0.8;
      }
    }
  }

  &.isByeBye {
    opacity: 0.5;

    .avator-icon {
      position: relative;

      &::after {
        content: "";
        width: inherit;
        height: 100%;
        position: absolute;
        top: 0px;
        left: 0px;
        z-index: 100;
        display: block;
        border: 2px solid var(--boke-color-danger);
        border-radius: inherit;
        box-sizing: border-box;
        pointer-events: none;
      }
    }
  }

  .left-icon {
    :deep(.el-avatar) {
      &::before {
        content: "";
        width: inherit;
        height: 100%;
        position: absolute;
        top: 0px;
        left: 4px;
        z-index: 100;
        display: block;
        border: 1px solid var(--boke-border-color-extra-light);
        border-radius: inherit;
        box-sizing: border-box;
        pointer-events: none;
        mix-blend-mode: darken;
      }
    }
  }

  :deep(.el-avatar) {
    img {
      width: 100%;
      cursor: pointer;
    }
  }
}

.demo-rich-conent {
  :deep(.el-avatar) {
    img {
      width: 100%;
      cursor: pointer;
    }
  }
}

.only-icon-style {
  padding: 0;
  margin: 0;
}
</style>
