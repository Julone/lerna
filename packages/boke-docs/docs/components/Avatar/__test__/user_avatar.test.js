import UserAvatar from '../demo/normal.vue'
import UserAvatarWith_provider from '../demo/with_provider.vue'
import { mount, VueWrapper } from '@vue/test-utils';
import { describe, expect, it, vi, test, } from 'vitest';

test('user avatar is normal', async () => {
    const wrapper = mount(UserAvatar, {
        },
    );
    expect(wrapper.element).toMatchSnapshot() // 匹配快照
})

test('with provider', async () => {
    const wrapper = mount(UserAvatarWith_provider, {
        }
    )
    expect(wrapper.element).toMatchSnapshot() // 匹配快照
})