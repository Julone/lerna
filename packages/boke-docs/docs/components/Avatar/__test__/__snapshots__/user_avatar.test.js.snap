// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`user avatar is normal 1`] = `
<div>
  <div
    data-v-7871c292=""
    style="display: inline-flex;"
  >
    <span
      class="ant-tag css-dev-only-do-not-override-16pw25h ant-tag-default boke-useravator-wrapper is_tag"
      data-v-7871c292=""
      style="padding: 2px 12px 2px 2px; border-radius: 50px;"
    >
      
      <div
        class="left-icon"
        data-v-7871c292=""
      >
        <span
          class="avator-icon ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-16pw25h"
          data-v-7871c292=""
          style="width: 28px; height: 28px; line-height: 28px; font-size: 18px;"
        >
          <img
            src="https://s1-imfile.feishucdn.com/static-resource/v1/v3_009p_b3b75461-971a-41e0-ac58-f755d34ef05g~?image_size=noop&cut_type=&quality=&format=png&sticker_format=.webp"
          />
        </span>
      </div>
      
      
      <div
        class="label"
        data-v-7871c292=""
      >
        <span
          class="realname"
          data-v-7871c292=""
        >
          Julone
        </span>
        <span
          class="id"
          data-v-7871c292=""
        >
          <span
            data-v-7871c292=""
          >
            运维部
          </span>
        </span>
      </div>
      <!--v-if-->
      <!--v-if-->
      
      
      
      <!---->
    </span>
  </div>
   无数据的时候 
  <div
    data-v-7871c292=""
    style="display: inline-flex;"
  >
    <span
      class="ant-tag css-dev-only-do-not-override-16pw25h ant-tag-default boke-useravator-wrapper is_tag"
      data-v-7871c292=""
      style="padding: 2px 12px 2px 2px; border-radius: 50px;"
    >
      
      <div
        class="left-icon"
        data-v-7871c292=""
      >
        <span
          class="avator-icon ant-avatar ant-avatar-circle css-dev-only-do-not-override-16pw25h"
          data-v-7871c292=""
          style="width: 28px; height: 28px; line-height: 28px; font-size: 18px;"
        >
          <span
            class="ant-avatar-string"
            style="opacity: 0;"
          >
            <!---->
          </span>
        </span>
      </div>
      
      
      <div
        class="label"
        data-v-7871c292=""
      >
        <span
          class="realname"
          data-v-7871c292=""
        >
          未知用户
        </span>
        <span
          class="id"
          data-v-7871c292=""
        >
          <span
            data-v-7871c292=""
          >
            未知部门
          </span>
        </span>
      </div>
      <!--v-if-->
      <!--v-if-->
      
      
      
      <!---->
    </span>
  </div>
</div>
`;

exports[`with provider 1`] = `
<div>
  <h4>
    Inside of Provider
  </h4>
  
  <div
    data-v-7871c292=""
    style="display: inline-flex;"
  >
    <span
      class="ant-tag css-dev-only-do-not-override-16pw25h ant-tag-default boke-useravator-wrapper is_tag is_highlight"
      data-v-7871c292=""
      style="padding: 2px 12px 2px 2px; border-radius: 50px;"
    >
      
      <div
        class="left-icon"
        data-v-7871c292=""
      >
        <span
          class="avator-icon ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-16pw25h"
          data-v-7871c292=""
          style="width: 28px; height: 28px; line-height: 28px; font-size: 18px;"
        >
          <img
            src="https://s1-imfile.feishucdn.com/static-resource/v1/v3_009p_b3b75461-971a-41e0-ac58-f755d34ef05g~?image_size=noop&cut_type=&quality=&format=png&sticker_format=.webp"
          />
        </span>
      </div>
      
      
      <div
        class="label"
        data-v-7871c292=""
      >
        <span
          class="realname"
          data-v-7871c292=""
        >
          Julone-new
        </span>
        <span
          class="id"
          data-v-7871c292=""
        >
          <span
            data-v-7871c292=""
          >
            运维部
          </span>
        </span>
      </div>
      <!--v-if-->
      <!--v-if-->
      
      
      
      <!---->
    </span>
  </div>
  
  <h4>
    Out of Provider
  </h4>
  <div
    data-v-7871c292=""
    style="display: inline-flex;"
  >
    <span
      class="ant-tag css-dev-only-do-not-override-16pw25h ant-tag-default boke-useravator-wrapper is_tag"
      data-v-7871c292=""
      style="padding: 2px 12px 2px 2px; border-radius: 50px;"
    >
      
      <div
        class="left-icon"
        data-v-7871c292=""
      >
        <span
          class="avator-icon ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-16pw25h"
          data-v-7871c292=""
          style="width: 28px; height: 28px; line-height: 28px; font-size: 18px;"
        >
          <img
            src="https://s1-imfile.feishucdn.com/static-resource/v1/v3_009p_b3b75461-971a-41e0-ac58-f755d34ef05g~?image_size=noop&cut_type=&quality=&format=png&sticker_format=.webp"
          />
        </span>
      </div>
      
      
      <div
        class="label"
        data-v-7871c292=""
      >
        <span
          class="realname"
          data-v-7871c292=""
        >
          Julone
        </span>
        <span
          class="id"
          data-v-7871c292=""
        >
          <span
            data-v-7871c292=""
          >
            运维部
          </span>
        </span>
      </div>
      <!--v-if-->
      <!--v-if-->
      
      
      
      <!---->
    </span>
  </div>
</div>
`;
